[Profiler] Regularize `AccumulateGrad` name

robieta opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #87568
* #87567
* #87566
* #87006
* #86880
* #86854
* #86853
* #86802
* #87133
* #87244
* __->__ #86909
* #86755
* #86754
* #86753

Memory profiler will use AccumulateGrad when detecting gradients. The name difference between Windows and other platforms has already cropped up with profiler trees so it makes sense to address it at the source.

Differential Revision: [D40347550](https://our.internmc.facebook.com/intern/diff/D40347550/)

Commit:
[Profiler] Regularize `AccumulateGrad` name

Memory profiler will use AccumulateGrad when detecting gradients. The name difference between Windows and other platforms has already cropped up with profiler trees so it makes sense to address it at the source.

Differential Revision: [D40347550](https://our.internmc.facebook.com/intern/diff/D40347550/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Regularize `AccumulateGrad` name"


Memory profiler will use AccumulateGrad when detecting gradients. The name difference between Windows and other platforms has already cropped up with profiler trees so it makes sense to address it at the source.

Differential Revision: [D40347550](https://our.internmc.facebook.com/intern/diff/D40347550/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Regularize `AccumulateGrad` name"


Memory profiler will use AccumulateGrad when detecting gradients. The name difference between Windows and other platforms has already cropped up with profiler trees so it makes sense to address it at the source.

Differential Revision: [D40347550](https://our.internmc.facebook.com/intern/diff/D40347550/)

[ghstack-poisoned]

slgong-fb commented on torch/csrc/profiler/collection.cpp:
```diff
@@ -296,6 +296,19 @@ void ThreadLocalSubqueue::TorchOpStorage::materialize(
+    auto& name = event.basic_fields_.name_;
+    auto position = name.find(windows_pattern);
+    if (position != std::string::npos) {
+      name.replace(position, windows_pattern.size(), accumulate_grad);
```
So we are just moving the replacement code into collection part?

robieta replied:
Exactly.

slgong-fb reviewed (approved):
LGTM!

Commit:
Update on "[Profiler] Regularize `AccumulateGrad` name"


Memory profiler will use AccumulateGrad when detecting gradients. The name difference between Windows and other platforms has already cropped up with profiler trees so it makes sense to address it at the source.

Differential Revision: [D40347550](https://our.internmc.facebook.com/intern/diff/D40347550/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Regularize `AccumulateGrad` name"


Memory profiler will use AccumulateGrad when detecting gradients. The name difference between Windows and other platforms has already cropped up with profiler trees so it makes sense to address it at the source.

Differential Revision: [D40347550](https://our.internmc.facebook.com/intern/diff/D40347550/)

[ghstack-poisoned]

robieta commented:
@pytorchbot merge

robieta commented:
@pytorchbot merge -f -m "Failures lower in the stack were unrelated flakes."

robieta commented:
@pytorchbot merge -f "Failures lower in the stack were unrelated flakes."

Commit:
Update on "[Profiler] Regularize `AccumulateGrad` name"


Memory profiler will use AccumulateGrad when detecting gradients. The name difference between Windows and other platforms has already cropped up with profiler trees so it makes sense to address it at the source.

Differential Revision: [D40347550](https://our.internmc.facebook.com/intern/diff/D40347550/)

[ghstack-poisoned]

aaronenyeshi commented on torch/csrc/profiler/collection.cpp:
```diff
@@ -296,6 +296,19 @@ void ThreadLocalSubqueue::TorchOpStorage::materialize(
+  // annotation relies on `c10::demangle` which is platform dependent. In
+  // particular, Windows will add a "struct " prefix.
+  const std::string accumulate_grad = "torch::autograd::AccumulateGrad";
+  const std::string windows_pattern = std::string("struct ") + accumulate_grad;
```
Is this pattern only important for AccumulateGrad, or should it be generalized elsewhere?

robieta replied:
It might crop up again, but in general it should be pretty limited. We generally try not to rely on demangled names; AccumulateGrad just happens to be very special.

aaronenyeshi reviewed (approved):
LGTM!

Commit:
Update on "[Profiler] Regularize `AccumulateGrad` name"


Memory profiler will use AccumulateGrad when detecting gradients. The name difference between Windows and other platforms has already cropped up with profiler trees so it makes sense to address it at the source.

Differential Revision: [D40347550](https://our.internmc.facebook.com/intern/diff/D40347550/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Regularize `AccumulateGrad` name"


Memory profiler will use AccumulateGrad when detecting gradients. The name difference between Windows and other platforms has already cropped up with profiler trees so it makes sense to address it at the source.

Differential Revision: [D40347550](https://our.internmc.facebook.com/intern/diff/D40347550/)

[ghstack-poisoned]

robieta commented:
@pytorchbot merge -f "the buck test failure is unrelated"
